module/VuFind/tests/integration-tests/src/VuFindTest/Mink/ApiTest.php

```diff
@@ -40,18 +47,106 @@
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
  * @link     https://vufind.org Main Page
  */
-class ApiTest extends \VuFindTest\Integration\MinkTestCase
+final class ApiTest extends \VuFindTest\Integration\MinkTestCase
 {
+    use \VuFindTest\Feature\LiveDatabaseTrait;
+    use \VuFindTest\Feature\LiveDetectionTrait;
+    use \VuFindTest\Feature\UserCreationTrait;
+    use \VuFindTest\Feature\DemoDriverTestTrait;
+    use \VuFindTest\Feature\EmailTrait;
+
+    /**
+     * Standard setup method.
+     *
+     * @return void
+     */
+    public static function setUpBeforeClass(): void
+    {
+        static::failIfDataExists();
+    }
+
+    /**
+     * Helper function to create a new API key entity
+     *
+     * @param string $title API key title
+     *
+     * @return ApiKeyEntityInterface
+     */
+    protected function getApiKey(string $title = 'test_title'): ApiKeyEntityInterface
+    {
+        $userService = $this->getLiveDbServiceManager()->get(UserServiceInterface::class);
+        $user = $userService->getUserByUsername('username1');
+        $developerSettingsService = $this->getLiveDatabaseContainer()->get(DeveloperSettingsService::class);
+        $apiKey = $developerSettingsService->generateApiKeyForUser($user, $title);
+        return $apiKey;
+    }
+
+    /**
+     * Helper function to get a revoked API key entity
+     *
+     * @return ApiKeyEntityInterface
+     */
+    protected function getRevokedApiKey(): ApiKeyEntityInterface
+    {
+        $apiKey = $this->getApiKey('fail_title');
+        $apiKey->setRevoked(true);
+        $apiKeyService = $this->getLiveDbServiceManager()->get(ApiKeyServiceInterface::class);
+        $apiKeyService->persistEntity($apiKey);
+        return $apiKey;
+    }
+
+    /**
+     * Helper function to set correct API key configs
+     *
+     * @param string $mode API key mode
+     *
+     * @return void
+     */
+    protected function setApiKeyConfigs(string $mode = 'disabled'): void
+    {
+        $this->changeConfigs(
+            [
+                'config' => [
+                    'API_Keys' => [
+                        'mode' => $mode,
+                        'token_salt' => 'test_token_salt',
+                        'key_limit' => 10,
+                    ],
+                    'Mail' => [
+                        'testOnly' => true,
+                        'message_log' => $this->getEmailLogPath(),
+                        'message_log_format' => $this->getEmailLogFormat(),
+                    ],
```

Comment:
Is there a reason for these mail settings? I don't see any tests related to emails.